mobilecoind/src/payments.rs

```diff
@@ -312,6 +312,76 @@ impl<T: UserTxConnection + 'static> TransactionsManager<T> {
         Ok(tx_proposal)
     }
 
+    pub fn generate_tx_from_tx_list(
+        &self,
+        account_key: &AccountKey,
+        input_list: &[UnspentTxOut],
+        receiver: &PublicAddress,
+        fee: u64,
+    ) -> Result<TxProposal, Error> {
+        let logger = self.logger.new(o!("receiver" => receiver.to_string()));
+        log::trace!(logger, "Generating txo list transaction...");
+
+        // All inputs are to be spent
+        let total_value: u64 = input_list.iter().map(|utxo| utxo.value).sum();
+        log::trace!(
+            logger,
+            "Total transaction value excluding fees: {}",
```

Comment:
This does not exclude the fee.

Reply:
There should be a check that total_value > fee.
If `fee` is 0, a default value should be used.